src/Microsoft.ML.DataView/VBuffer.cs

```diff
@@ -331,5 +333,76 @@ internal VBufferEditor<T> GetEditor()
                 createdNewValues,
                 createdNewIndices);
         }
+
+        /// <summary>
+        /// A helper method that gives us an iterable over the items given the fields from a <see cref="VBuffer{T}"/>.
+        /// Note that we have this in a separate utility class, rather than in its more natural location of
+        /// <see cref="VBuffer{T}"/> itself, due to a bug in the C++/CLI compiler. (DevDiv 1097919:
+        /// [C++/CLI] Nested generic types are not correctly imported from metadata). So, if we want to use
+        /// <see cref="VBuffer{T}"/> in C++/CLI projects, we cannot have a generic struct with a nested class
+        /// that has the outer struct type as a field.
+        /// </summary>
+        private static IEnumerable<KeyValuePair<int, T>> Items(T[] values, int[] indices, int length, int count, bool all)
+        {
+            Debug.Assert(0 <= count && count <= ArrayUtils.Size(values));
+            Debug.Assert(count <= length);
+            Debug.Assert(count == length || count <= ArrayUtils.Size(indices));
```

Comment:
>Assert [](start = 18, length = 6)

should this also check the size of indices? Or can indices be an empty array (like when count == length)?

Reply:
Indices can be an empty array - actually `null`. This is useful when it is a dense VBuffer (when count == length), because then you don't need to allocate a useless array.